### ISSUE TITLE
Validate DKIM with CNAME

### ISSUE THREAD

**Issue by ANICHKAR** [atech/postal#868]
I'd like to know if does anyone know how to make the app/models/domain/dns_checks.rb to validate DKIM with a CNAME pointing to another domain which holds a valid selector with a public key? Currently it only validates TXT.

**Comment by willpower232** [comment]
This is possible if you modify the code however the DKIM record should be unique to the domain name in Postal and is also verified by third parties who receive email from Postal so it doesn't make much sense to reference another record with CNAME and is likely to cause more problems.

**Comment by ANICHKAR** [comment]
Thank you @willpower232, but for me at least it makes sense. It is a way to make things easier and avoid mistakes.  It is easier for noobs to create only CNAME records than create MX, TXT and CNAME. I'm integrating with Cloudflare API, so those records are going to be created automatically.

https://sendgrid.com/docs/ui/account-and-settings/how-to-set-up-domain-authentication/#using-automated-security

**Comment by willpower232** [comment]
I understand your use case now. Presumably its a valid way of making it work if Sendgrid offer it.

In theory, you could swap `Resolv::DNS::Resource::IN::TXT` for `Resolv::DNS::Resource::IN::ANY` and it would check all kinds of records matching the name given so as long as your customers use `postal-whatever._domainkey.theirdomain.com`, it might work.

**Comment by ANICHKAR** [comment]
I changed, but now it returns an error 500. I think the problem is related to the next line: 

> records = result.map(&:data)

**Comment by alnalda** [comment]
Hi, i'm facing with almost the same issue that @ANICHKAR had. 

I'm using CNAME records pointing to the real TXT dkim and the issue i have is that, there are multiple domains that are recognised and others no, ie. some of them are marked as valid and other as invalid.

I already checked the records just in case and are exactly the same. I also ran the nslookup from the server and the responses are the same finding the records with the CNAME line and the TXT one.

I didn't change the code behind yet so the query is asking for TXT records. 

Has anybody this issue? 
Thanks.
